java/adapter/avro/src/main/java/org/apache/arrow/consumers/CompositeAvroConsumer.java

```diff
@@ -53,8 +53,12 @@ public void consume(Decoder decoder, VectorSchemaRoot root) throws IOException {
    * Reset vector of consumers with the given {@link VectorSchemaRoot}.
    */
   public void resetConsumerVectors(VectorSchemaRoot root) {
-    for (int i = 0; i < root.getFieldVectors().size(); i++) {
-      consumers.get(i).resetValueVector(root.getFieldVectors().get(i));
+    int index = 0;
+    for (Consumer consumer : consumers) {
+      if (consumer instanceof SkippableConsumer) {
+        continue;
+      }
+      consumer.resetValueVector(root.getFieldVectors().get(index++));
```

Comment:
it might be cleaner to have this return a boolean to indicate if the field should be advanced instead of checking for a specific class type.